### PR TITLE
Set the type and immutability of the secret when copying the self-signed-cert to the user's namespace.

### PR DESCRIPTION
**Pull request by metlos**
### What does this PR do?
This makes sure that the `type` and `immutability` is set on the secret representing the che self signed certificate in the user's namespace.

### Screenshot/screencast of this PR
N/A

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/20546

### How to test this PR?
1. Use OpenShift
1. `chectl server:deploy -p openshift -a operator --workspace-engine=dev-workspace`
2. Create a namespace/project and label it with  `che.eclipse.org/workspace-namespace-owner-uid` with value of the UID of some user.
3. Create a self-signed certificate in the namespace where there is the `eclipse-che` custom resource (usually `eclipse-che` namespace)
4. There should be a new secret created in the new namespace with the copy of `ca.crt` of the self-signed-certificate from the eclipse-che namespace.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [Custom resource definition file is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#updating-custom-resource-definition-file)
- [x] [OLM bundles are up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#update-olm-bundle)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: bundle/next-all-namespaces/eclipse-che-preview-openshift/metadata/annotations.yaml [hunk 1]
```diff
@@ -13,5 +13,4 @@ annotations:
   # Annotations for testing.
   operators.operatorframework.io.test.mediatype.v1: scorecard+v1
   operators.operatorframework.io.test.config.v1: tests/scorecard/
-
-  com.redhat.openshift.versions: "v4.8"
+\n  com.redhat.openshift.versions: "v4.8"
```
Comment (skabashnyuk) [inline bundle/next-all-namespaces/eclipse-che-preview-openshift/metadata/annotations.yaml:16]: looks irreleveant. No?

Comment (tolusha) [inline bundle/next-all-namespaces/eclipse-che-preview-openshift/metadata/annotations.yaml:16]: I think it is autogenerated. (probably bug in our scripts)
But could you revert this?

Comment (metlos) [reply bundle/next-all-namespaces/eclipse-che-preview-openshift/metadata/annotations.yaml:16]: It does, but I have no control over this - this is generated during the `make update-resources` goal...

**Review by tolusha — APPROVED** [review]

**Pull request merged**